File: 121/121.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let low = prices[0]
    let high = 0
    prices.forEach((price, indx) => {
      low = Math.min(low, price)
      high = Math.max(high, price-low)
    })
    return high
};
res = maxProfit([7, 1, 5, 3, 6, 4])
console.log(res)

File: 121/121.go
package main

import "fmt"

func maxProfit(prices []int) int {
	minPrice := 10001
	profit := 0

	for _, price := range prices {
		minPrice = min(minPrice, price)
		profit = max(profit, price-minPrice)
	}
	return profit
}

func main() {
	fmt.Println(maxProfit([]int{7, 1, 5, 3, 6, 4}))
}
